### ISSUE TITLE
Suppress CS4014 when `.Received()` is used

### ISSUE THREAD

**Issue by aradalvand** [nsubstitute/NSubstitute.Analyzers#220]
Currently, the following:
```cs
sub.Received().DoSomethingAsync();
```
results in the CS4014 warning being shown saying that the call should be `await`ed:
> Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.

but it doesn't actually need to be awaited, so `NSubstitute.Analyzers` could override this warning in cases like this.

**Comment by tpodolak** [comment]
`CS4014` is compiler warning, `NSubstitute.Analyzers` cannot suppress these warnings
